project_event/models/project_task.py

```diff
@@ -282,12 +282,43 @@ def create_new_activity(self, vals):
             .next_by_code('project.task.activity')
         return super(Task, self).create(vals)
 
+    @api.multi
+    def update_reservation_event(self, update_task):
+        if update_task and self.reservation_event_id:
+            self.ensure_one()
+            reservation_event = self.env['calendar.event'].browse(
+                self.reservation_event_id)
+            partners = []
+            for e in self.employee_ids:
+                if e.user_id:
+                    partners.append(e.user_id.partner_id.id)
+                else:
+                    raise UserError(
+                        _('Please define user account for the %s employee') % (
+                            e.name,))
+            reservation_event.write(
+                {
+                    'start': self.date_start,
+                    'stop': self.date_end,
+                    'name': self.name,
+                    'resource_type': self.resource_type,
+                    'room_id': self.room_id.id if self.room_id else None,
+                    'equipment_ids': [(6, 0,
+                                       self.get_equipment_ids_inside())] if self.room_id
+                    else [(6, 0, self.equipment_id.id)],
+                    'partner_ids': [(6, 0, partners)],
+                    'category_id': self.category_id.id,
+                }
+            )
+
     @api.multi
     def write(self, vals):
         if self.activity_task_type == 'activity':
             return self.write_activity(vals)
         else:
-            return super(Task, self).write(vals)
+            update_task = super(Task, self).write(vals)
+            self.update_reservation_event(update_task)
```

Comment:
why you pass update_task? I think it should be either self since self=update_task or vals if you want to pass only the changed values